src/sentry/api/serializers/models/releaseactivity.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping, Sequence, TypedDict
+
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import Group
+from sentry.models.releaseactivity import ReleaseActivity
+from sentry.utils.json import JSONData
+
+
+class ReleaseActivityResponse(TypedDict):
+    id: int
+    type: str
+    data: Mapping[str, JSONData]
+    date_added: datetime
+
+
+@register(ReleaseActivity)
+class ReleaseActivitySerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[ReleaseActivity], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        def _update_data(d: MutableMapping[str, Any]):
+            if d.get("group_id"):
+                try:
+                    d["group"] = Group.objects.get(id=d.get("group_id"))
```

Comment:
This is still going to make N queries to fetch each group. You want to fetch all the group ids and get them all in one bulk query.

Reply:
Actually just to build on that a bit - you might want to fetch all groups, `serialize` them in one bulk call and then assign them to their respective activity